services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableIndexManager.java

```diff
@@ -187,6 +191,19 @@ public Long populateViewFromEntityReplication(Long tableId, ProgressCallback cal
 	 * @return A ColumnModel for each distinct annotation for the given scope.
 	 */
 	public ColumnModelPage getPossibleColumnModelsForView(Long viewId, String nextPageToken);
+
+	/**
+	 * Build the index for the given table using the provided change metadata up to and
+	 * including the provided last change number.
+	 * @param progressCallback 
+	 * 
+	 * @param tableId
+	 * @param iterator
+	 * @param lastChangeNumber
+	 * @throws RecoverableMessageException Will RecoverableMessageException if the index cannot be built at this time.
+	 */
+	public void buildIndexToChangeNumber(ProgressCallback progressCallback, IdAndVersion idAndVersion, Iterator<TableChangeMetaData> iterator,
+			Optional<Long> lastChangeNumber) throws RecoverableMessageException;

```

Comment:
lastChangeNumber is not actually optional.   Force the caller to not call this method if the value is they have is null.